src/Shared/TaskFactoryUtilities.cs

```diff
@@ -0,0 +1,263 @@
+
+            foreach (string assemblyPath in assemblyPaths)
+            {
+                if (!string.IsNullOrEmpty(assemblyPath) && File.Exists(assemblyPath))
+                {
+                    string? directory = Path.GetDirectoryName(assemblyPath);
+                    if (!string.IsNullOrEmpty(directory) && seenDirectories.Add(directory))
+                    {
+                        directories.Add(directory);
+                    }
+                }
+            }
+
+            return directories;
+        }
+
+        /// <summary>
+        /// Loads an assembly based on whether out-of-process execution is enabled.
+        /// </summary>
+        /// <param name="assemblyPath">The path to the assembly to load.</param>
+        /// <param name="useOutOfProcess">Whether to use out-of-process execution mode.</param>
+        /// <returns>The loaded assembly.</returns>
+        public static Assembly LoadTaskAssembly(string assemblyPath, bool useOutOfProcess)
+        {
+            if (string.IsNullOrEmpty(assemblyPath))
+            {
+                throw new ArgumentException("Assembly path cannot be null or empty.", nameof(assemblyPath));
+            }
+
+            if (useOutOfProcess)
+            {
+                return Assembly.LoadFrom(assemblyPath);
+            }
+            else
+            {
+                return Assembly.Load(File.ReadAllBytes(assemblyPath));
+            }
+        }
+
+        /// <summary>
+        /// Registers assembly resolution handlers for inline tasks based on their load manifest file.
+        /// This enables out-of-process task execution to resolve dependencies that were identified
+        /// during TaskFactory initialization.
+        /// </summary>
+        /// <param name="taskLocation">The path to the task assembly.</param>
+        /// <returns>True if a manifest was found and handlers were registered, false otherwise.</returns>
+        public static bool RegisterAssemblyResolveHandlersFromManifest(string taskLocation)
+        {
+            if (string.IsNullOrEmpty(taskLocation))
+            {
+                throw new ArgumentException("Task location cannot be null or empty.", nameof(taskLocation));
+            }
+
+            if (!taskLocation.EndsWith(InlineTaskSuffix, StringComparison.OrdinalIgnoreCase))
+            {
+                return false;
+            }
+
+            string manifestPath = taskLocation + InlineTaskLoadManifestSuffix;
+            if (!File.Exists(manifestPath))
+            {
+                return false;
+            }
+
+            string[] directories = File.ReadAllLines(manifestPath);
+            if (directories?.Length > 0)
+            {
+                var resolver = CreateAssemblyResolver(new List<string>(directories));
+                AppDomain.CurrentDomain.AssemblyResolve += resolver;
+                return true;
+            }
+
+            return false;
```

Comment:
nit: I think it would be more logical that if the manifest file is empty (they should not exist, but still), the function returns true - empty set of handlers was registered successfully. Or we can return the amount instead of boolean.

Reply:
the return value is actually not used so I removed it